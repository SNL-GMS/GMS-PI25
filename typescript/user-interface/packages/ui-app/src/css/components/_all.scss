@import './popout-component.scss';
@import './about-menu.scss';
@import './body.scss';
@import './command-palette.scss';
@import './list.scss';
@import './login.scss';
@import './quadra-grid.scss';
@import './status-bar.scss';
@import './theme-provider.scss';
@import './workspace.scss';
