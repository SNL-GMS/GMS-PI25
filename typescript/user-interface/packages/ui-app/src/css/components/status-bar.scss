.status-bar {
  background-color: var(--gms-table-header-background, $gms-table-header-background);
  border-top: 1px solid var(--gms-recessed, $gms-recessed);
  bottom: 0;
  display: flex;
  font-size: 1rem;
  font-weight: 300;
  gap: 0.5em;
  height: calc(1.75rem - 1px); // handle border size
  letter-spacing: 1px;
  position: relative;
  width: 100%;
  z-index: 1;
  @include addBoxShadow('lg');
}

.loading-counter {
  align-items: center;
  border-radius: 0.25rem;
  display: flex;
  font-family: $gms-mono;
  justify-content: center;
}

.loading-widget {
  align-items: center;
  display: flex;
  gap: 0.5rem;
  height: 100%;
  padding: 0 0.5rem;
  position: relative;

  &:hover {
    cursor: pointer;
    @include castLightOn(0.1);
  }

  &:hover .bp4-icon-issue {
    color: red;
  }
}

.loading-info {
  font-size: 1rem;
  font-weight: 300;
  letter-spacing: 1px;

  &__label {
    font-weight: 500;
  }

  &__loading-info {
    display: flex;
    font-size: 0.875rem;
    gap: 0.5rem;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}

.loading-list {
  --gms-scrollbar-background: lightgray;
  --gms-scrollbar: gray;
  display: flex;
  flex-direction: column-reverse;
  gap: 0.25rem;
  list-style-type: none;
  margin-block-end: 0;
  margin-block-start: 0;
  margin-inline-end: 0;
  margin-inline-start: 0;
  max-height: 30em;
  min-width: 40ch;
  overflow-y: auto;
  padding-inline-start: 0;
  padding: 0.5rem 0.5rem 0 0;

  &__sticky-label {
    border-bottom: 1px solid var(--gms-main-inverted, $gms-main-inverted);
    font-size: 1.25em;
    font-weight: 600;
    padding: 0.25rem 0.5rem 0.25rem 0;
    position: sticky;
    top: 0;
  }
}

.loading-list-entry {
  position: relative;

  &__description {
    padding-left: 1.25rem;
    position: relative;

    &--set-apart {
      padding-bottom: 0.25rem;
      padding-top: 0.25rem;
    }
  }

  &__label {
    font-weight: 600;
  }

  &.is-error {
    color: darkred;
  }

  &__icon,
  &__spinner {
    left: 0;
    position: absolute;
    top: 50%;
    transform: translate3D(0, -50%, 0);
  }

  &__more-info {
    display: block;
    padding-left: 1.25rem;
  }
}
