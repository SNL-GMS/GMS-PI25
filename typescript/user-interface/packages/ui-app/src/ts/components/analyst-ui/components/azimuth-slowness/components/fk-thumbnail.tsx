/* eslint-disable react/destructuring-assignment */
import type { FkTypes } from '@gms/common-model';
import { useColorMap } from '@gms/ui-state';
import isEqual from 'lodash/isEqual';
import React, { useEffect, useState } from 'react';

import * as fkUtil from './fk-util';

/**
 * Fk Thumbnail Props.
 */
export interface FkThumbnailProps {
  fkData: FkTypes.FkPowerSpectra;
  sizePx: number;
  label: string;
  fkUnit: FkTypes.FkUnits;
  dimFk: boolean;
  highlightLabel?: boolean;
  predictedPoint?: FkTypes.AzimuthSlowness;
  selected?: boolean;
  arrivalTimeMovieSpectrumIndex: number;

  showFkThumbnailMenu?(event: React.MouseEvent): void;
  onClick?(e: React.MouseEvent<HTMLCanvasElement>): void;
}

export function FkThumbnail(props: FkThumbnailProps) {
  /** destination to draw the fk. */
  let canvasRef: HTMLCanvasElement | undefined;

  /** The current fk represented as an ImageBitmap. */
  const currentImage = React.useRef<ImageBitmap>();

  const classNames = ['fk-thumbnail', props.selected ? 'selected' : undefined].join(' ');
  const [colorMap] = useColorMap();

  /**
   * Displays a context menu for reviewing/clearing an fk
   */
  function showThumbnailContextMenu(e: React.MouseEvent<HTMLDivElement>) {
    e.preventDefault();
    if (props.showFkThumbnailMenu) {
      props.showFkThumbnailMenu(e);
    }
  }

  const [currentFkDisplayData, setCurrentFkDisplayData] = useState<number[][]>([]);
  const [prevProps, setPreviousProps] = useState<FkThumbnailProps>(undefined);

  useEffect(() => {
    async function fetchCreateImage(fkDisplayData: number[][]) {
      const [min, max] = fkUtil.computeMinMaxFkValues(fkDisplayData);
      currentImage.current = await fkUtil.createFkImageBitmap(fkDisplayData, min, max, colorMap);
      fkUtil.drawImage(canvasRef, currentImage.current, props.dimFk);
      fkUtil.draw(
        canvasRef,
        props.fkData,
        props.predictedPoint,
        props.arrivalTimeMovieSpectrumIndex,
        true,
        props.dimFk
      );
      setCurrentFkDisplayData(fkDisplayData);
      setPreviousProps(props);
    }

    if (!props.fkData || !props.fkData.values) {
      return;
    }

    const fkDisplayData = fkUtil.getFkHeatmapArrayFromFkSpectra(
      props.fkData.values[props.arrivalTimeMovieSpectrumIndex],
      props.fkUnit
    );

    if (fkDisplayData.length === 0) {
      return;
    }

    if (
      !isEqual(currentFkDisplayData, fkDisplayData) ||
      props.fkUnit !== prevProps?.fkUnit ||
      props.dimFk !== prevProps?.dimFk
    ) {
      // eslint-disable-next-line @typescript-eslint/no-floating-promises
      fetchCreateImage(fkDisplayData);
    } else if (currentImage && prevProps?.sizePx !== props.sizePx) {
      fkUtil.drawImage(canvasRef, currentImage.current, props.dimFk);
      fkUtil.draw(
        canvasRef,
        props.fkData,
        props.predictedPoint,
        props.arrivalTimeMovieSpectrumIndex,
        true,
        props.dimFk
      );
      setPreviousProps(props);
    }
  }, [
    props,
    currentFkDisplayData,
    prevProps?.fkUnit,
    prevProps?.dimFk,
    prevProps?.sizePx,
    canvasRef,
    colorMap
  ]);

  const createOnClick = React.useCallback(
    (e: React.MouseEvent<HTMLCanvasElement>) => {
      if (props.onClick) {
        props.onClick(e);
      }
    },
    [props]
  );
  return (
    <div
      className={classNames}
      style={{
        width: `${props.sizePx}px`,
        height: `${props.sizePx}px`
      }}
      onContextMenu={e => showThumbnailContextMenu(e)}
    >
      <div
        className={
          !props.highlightLabel || props.dimFk
            ? 'fk-thumbnail__label--reviewed'
            : 'fk-thumbnail__label'
        }
      >
        {props.label}
      </div>
      <canvas
        className="fk-thumbnail__canvas"
        height={props.sizePx}
        width={props.sizePx}
        ref={ref => {
          canvasRef = ref;
        }}
        onClick={createOnClick}
      />
    </div>
  );
}
