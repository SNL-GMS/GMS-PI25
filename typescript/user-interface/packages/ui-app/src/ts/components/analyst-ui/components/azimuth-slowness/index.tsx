import { IanDisplays, toDisplayTitle } from '@gms/common-model/lib/displays/types';
import { nonIdealStateWithSpinner } from '@gms/ui-core-components';
import React from 'react';

import type { AzimuthSlownessProps } from './types';

const AzimuthSlownessLazy = React.lazy(async () =>
  import(
    /* webpackChunkName: 'ui-app-azimuth-slowness' */ './azimuth-slowness-container'
  ).then(module => ({ default: module.AzimuthSlowness }))
);

export function AzimuthSlowness(props: AzimuthSlownessProps) {
  return (
    <React.Suspense
      fallback={nonIdealStateWithSpinner(
        toDisplayTitle(IanDisplays.AZIMUTH_SLOWNESS),
        'Loading Display'
      )}
    >
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <AzimuthSlownessLazy {...props} />
    </React.Suspense>
  );
}
