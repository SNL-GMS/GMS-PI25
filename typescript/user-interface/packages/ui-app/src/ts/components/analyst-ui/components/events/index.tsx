import { IanDisplays, toDisplayTitle } from '@gms/common-model/lib/displays/types';
import { nonIdealStateWithSpinner } from '@gms/ui-core-components';
import React from 'react';

import type { EventsComponentProps } from './events-component';

const EventsLazy = React.lazy(async () =>
  import(/* webpackChunkName: 'ui-app-events' */ './events-component').then(module => ({
    default: module.EventsComponent
  }))
);

export function Events(props: EventsComponentProps) {
  return (
    <React.Suspense
      fallback={nonIdealStateWithSpinner(toDisplayTitle(IanDisplays.EVENTS), 'Loading Display')}
    >
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <EventsLazy {...props} />
    </React.Suspense>
  );
}
