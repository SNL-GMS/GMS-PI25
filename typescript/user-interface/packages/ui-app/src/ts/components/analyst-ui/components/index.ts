export { AzimuthSlowness } from './azimuth-slowness';
export { Events } from './events';
export { Filters } from './filters';
export { History } from './history';
export { Location } from './location';
export { Magnitude } from './magnitude';
export { IANMap } from './map';
export { SignalDetections } from './signal-detections';
export { StationProperties } from './station-properties';
export { Waveform } from './waveform';
export { Workflow } from './workflow';
