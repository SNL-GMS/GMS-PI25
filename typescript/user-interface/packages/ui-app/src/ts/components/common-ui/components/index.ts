export { CommandPalette } from './command-palette';
export { SystemMessage } from './system-message';
