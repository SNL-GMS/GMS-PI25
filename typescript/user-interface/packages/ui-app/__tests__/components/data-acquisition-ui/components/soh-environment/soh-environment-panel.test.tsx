/* eslint-disable react/jsx-no-constructed-context-values */
import { SohTypes } from '@gms/common-model';
import { uuid } from '@gms/common-util';
import { getStore } from '@gms/ui-state';
import uniqueId from 'lodash/uniqueId';
import React from 'react';
import { Provider } from 'react-redux';

import { BaseDisplayContext } from '../../../../../src/ts/components/common-ui/components/base-display';
import type { EnvironmentPanelProps } from '../../../../../src/ts/components/data-acquisition-ui/components/soh-environment/soh-environment-panel';
import { EnvironmentPanel } from '../../../../../src/ts/components/data-acquisition-ui/components/soh-environment/soh-environment-panel';
import type { SohContextData } from '../../../../../src/ts/components/data-acquisition-ui/shared/soh-context';
import { SohContext } from '../../../../../src/ts/components/data-acquisition-ui/shared/soh-context';
import type { FilterableSOHTypes } from '../../../../../src/ts/components/data-acquisition-ui/shared/toolbars/soh-toolbar';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

uuid.asString = jest.fn().mockImplementation(uniqueId);

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('SohEnvironmentPanel class', () => {
  const channel: SohTypes.ChannelSoh[] = [
    {
      allSohMonitorValueAndStatuses: [
        {
          status: SohTypes.SohStatusSummary.GOOD,
          value: 1,
          valuePresent: true,
          monitorType: SohTypes.SohMonitorType.ENV_ZEROED_DATA,
          hasUnacknowledgedChanges: false,
          contributing: true,
          thresholdMarginal: 1,
          thresholdBad: 10,
          quietUntilMs: 1
        }
      ],
      channelName: 'channelName',
      channelSohStatus: SohTypes.SohStatusSummary.GOOD
    }
  ];
  const channelStatusesToDisplay: Record<FilterableSOHTypes, boolean> = {
    NONE: false,
    BAD: false,
    GOOD: true,
    MARGINAL: false
  };
  const monitorStatusesToDisplay: Record<SohTypes.SohStatusSummary, boolean> = {
    NONE: false,
    BAD: false,
    GOOD: true,
    MARGINAL: false
  };
  const myProps: EnvironmentPanelProps = {
    channelSohs: channel,
    sohStatusContributors: [],
    channelStatusesToDisplay,
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    quietingDurationSelections: [1, 5, 10],
    defaultQuietDurationMs: 10,
    monitorStatusesToDisplay,
    stationName: 'AAK',
    isStale: false
  };

  const contextDefaults: SohContextData = {
    glContainer: {} as any,
    selectedAceiType: SohTypes.AceiType.BEGINNING_TIME_OUTAGE,
    setSelectedAceiType: jest.fn()
  };

  const panelContent = (
    <Provider store={getStore()}>
      <BaseDisplayContext.Provider
        value={{
          glContainer: { width: 150, height: 150 } as any,
          widthPx: 150,
          heightPx: 150
        }}
      >
        <SohContext.Provider value={contextDefaults}>
          {/* eslint-disable-next-line react/jsx-props-no-spreading */}
          <EnvironmentPanel {...myProps} />
        </SohContext.Provider>
      </BaseDisplayContext.Provider>
    </Provider>
  );
  const sohEnvironmentPanel = Enzyme.mount(panelContent);
  const sohShallowEnvironmentPanel = Enzyme.shallow(panelContent);
  const sohEnvironmentPanel2: any = new EnvironmentPanel(myProps);

  it('should be defined', () => {
    expect(sohEnvironmentPanel).toBeDefined();
    expect(sohEnvironmentPanel2).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(sohShallowEnvironmentPanel).toMatchSnapshot();
  });

  it('can use value getters', () => {
    const monitorTypeResult = sohEnvironmentPanel
      .find(EnvironmentPanel)
      .instance()
      .monitorTypeValueGetter({
        data: { id: SohTypes.SohMonitorType.ENV_ZEROED_DATA }
      });
    const channelValueResult = sohEnvironmentPanel
      .find(EnvironmentPanel)
      .instance()
      .channelValueGetter({
        data: {
          id: SohTypes.SohMonitorType.ENV_ZEROED_DATA
        },
        colDef: {
          colId: 'channelName'
        }
      });
    expect(monitorTypeResult).toEqual(SohTypes.SohMonitorType.ENV_ZEROED_DATA);
    expect(channelValueResult).toEqual(1);
  });

  it('can handle isMonitorTypeColumn', () => {
    const params = {
      colDef: {
        colId: SohTypes.SohMonitorType.ENV_ZEROED_DATA
      },
      event: {
        metaKey: true,
        ctrlKey: true
      },
      data: {
        id: SohTypes.SohMonitorType.ENV_ZEROED_DATA
      }
    };
    const result = sohEnvironmentPanel
      .find(EnvironmentPanel)
      .instance()
      .isMonitorTypeColumn(params);
    expect(result).toBeFalsy();
  });

  it('can handle onCellClicked', () => {
    const props = {
      colDef: {
        colId: 'Monitor Type'
      },
      data: {
        id: SohTypes.SohMonitorType.ENV_ZEROED_DATA
      },
      event: {
        metaKey: true
      }
    };
    sohEnvironmentPanel.find(EnvironmentPanel).instance().onCellClicked(props);
    expect(sohEnvironmentPanel).toBeDefined();
  });

  it('can handle onCellContextMenu', () => {
    const props = {
      colDef: {
        colId: 'Monitor Type'
      },
      data: {
        id: SohTypes.SohMonitorType.ENV_ZEROED_DATA
      },
      event: {
        metaKey: true
      }
    };
    sohEnvironmentPanel.find(EnvironmentPanel).instance().onCellContextMenu(props);
    expect(sohEnvironmentPanel).toBeDefined();
  });

  it('can handle onCellContextMenu not monitor type', () => {
    const props = {
      colDef: {
        colId: 'test'
      },
      data: {
        id: SohTypes.SohMonitorType.ENV_ZEROED_DATA
      },
      event: {
        metaKey: true
      }
    };
    sohEnvironmentPanel.find(EnvironmentPanel).instance().onCellContextMenu(props);
    expect(sohEnvironmentPanel).toBeDefined();
  });
  it('can handle componentDidUpdate', () => {
    sohEnvironmentPanel.find(EnvironmentPanel).instance().componentDidUpdate();
    expect(sohEnvironmentPanel).toBeDefined();
  });

  it('can handle selectCells', () => {
    const params = {
      params: {
        colDef: {
          colId: 'Monitor Type'
        },
        event: {
          metaKey: true,
          ctrlKey: true
        },
        data: {
          id: SohTypes.SohMonitorType.ENV_ZEROED_DATA
        }
      },
      shouldRemoveIfExisting: true,
      callback: jest.fn()
    };
    sohEnvironmentPanel
      .find(EnvironmentPanel)
      .instance()
      .selectCells(params.params, params.shouldRemoveIfExisting, params.callback);
    expect(sohEnvironmentPanel).toBeDefined();
  });
});
