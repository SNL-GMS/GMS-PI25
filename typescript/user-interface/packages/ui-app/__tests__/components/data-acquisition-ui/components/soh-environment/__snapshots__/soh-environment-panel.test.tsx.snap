// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SohEnvironmentPanel class should match snapshot 1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "store": {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ContextProvider
    value={
      {
        "glContainer": {
          "height": 150,
          "width": 150,
        },
        "heightPx": 150,
        "widthPx": 150,
      }
    }
  >
    <ContextProvider
      value={
        {
          "glContainer": {},
          "selectedAceiType": "BEGINNING_TIME_OUTAGE",
          "setSelectedAceiType": [MockFunction],
        }
      }
    >
      <EnvironmentPanel
        channelSohs={
          [
            {
              "allSohMonitorValueAndStatuses": [
                {
                  "contributing": true,
                  "hasUnacknowledgedChanges": false,
                  "monitorType": "ENV_ZEROED_DATA",
                  "quietUntilMs": 1,
                  "status": "GOOD",
                  "thresholdBad": 10,
                  "thresholdMarginal": 1,
                  "value": 1,
                  "valuePresent": true,
                },
              ],
              "channelName": "channelName",
              "channelSohStatus": "GOOD",
            },
          ]
        }
        channelStatusesToDisplay={
          {
            "BAD": false,
            "GOOD": true,
            "MARGINAL": false,
            "NONE": false,
          }
        }
        defaultQuietDurationMs={10}
        isStale={false}
        monitorStatusesToDisplay={
          {
            "BAD": false,
            "GOOD": true,
            "MARGINAL": false,
            "NONE": false,
          }
        }
        quietingDurationSelections={
          [
            1,
            5,
            10,
          ]
        }
        sohStatusContributors={[]}
        stationName="AAK"
      />
    </ContextProvider>
  </ContextProvider>
</ContextProvider>
`;
